packages/discord.js/typings/index.d.ts

```diff
@@ -5232,7 +5250,7 @@ export interface ClientEvents {
   guildUpdate: [oldGuild: Guild, newGuild: Guild];
   inviteCreate: [invite: Invite];
   inviteDelete: [invite: Invite];
-  messageCreate: [message: Message];
+  messageCreate: [message: Message & { channel: Exclude<TextBasedChannel, PartialGroupDMChannel> }];
```

Comment:
The same should be done for `messageDelete`, `messageUpdate`, `messageDeleteBulk` and `messageReactionRemoveAll` events.